File: 2486-most-frequent-even-element/most-frequent-even-element.cpp
class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
	    int ans = -1;
		
		//vector to find the maximum frequency of the elements
        vector<int> findMax;
		
		//vector to store the frequency of even numbers only
        vector<int> even;
		
		//hashing
        unordered_map<int,int> mp;
        for(int i=0;i<nums.size();++i){
            if(nums[i]%2==0){
            mp[nums[i]]++;
            }
        }
		
		//finding maximum frequency
        int max=0;
        for(auto i: mp){
           findMax.push_back(i.second); 
        }
		
		//if there is no even number in the array then its frequency will also be null hence return -1
        if(findMax.size()==0) return ans;
		
        sort(findMax.begin(),findMax.end());
        max=findMax[findMax.size()-1];
		
		
		//as we need to return the even number with max frequency , hence store only those numbers 
		// whose frequency is equal to max
        for(auto i:mp){
            if(i.second==max){
                even.push_back(i.first);
            }
        }
		
		//sorting the even numbers because if the frequency of any number of elements is same 
		// then we need to return the smallest among those 
        sort(even.begin(),even.end());
       
        if(even.size()>=1) return even[0];
        else  return ans;
    }
};

File: 2486-most-frequent-even-element/most-frequent-even-element.py
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        d={}
        for i in nums:
            if i%2==0:
                d[i]=d.get(i,0)+1
        s=0
        for i in d.values():
            if s<i:
                s=i
        res_even=[]
        for i,j in d.items():
            if j==s:
                res_even.append(i)
        res_even.sort()
        if len(res_even)>=1:
            return res_even[0]
        else:
            return -1